### PR TITLE
find the substrings for give string

### PR DESCRIPTION
**Pull request by Jeko0**

### CODE REVIEW

File: week5_sub_strings/lib/sub_strings.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    e.message
+  end
+
+  private
+
+  def calculate_sub_strings
+    substrings = Hash.new(0)
+    user_input = @sentence.split(" ")
+    user_input.each do |userword|
+      @dictionary.each do |word|
```
Comment (alexshvedko) [inline week5_sub_strings/lib/sub_strings.rb]: move nested loop to new methods

File: week5_sub_strings/lib/sub_strings.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        substrings[word] += 1 if userword.downcase.include?(word)
+      end
+    end
+    substrings
+  end
+
+  def check_input
+    raise InvalidinputError, "invalid input" unless @sentence.is_a?(String) && @dictionary.all?{|elemet| elemet.is_a?(String)}
```
Comment (alexshvedko) [inline week5_sub_strings/lib/sub_strings.rb]: long line

File: week5_sub_strings/lib/sub_strings.rb (hunk 1 above)
Comment (Jeko0) [reply week5_sub_strings/lib/sub_strings.rb]: removed double nesting

File: week5_sub_strings/lib/sub_strings.rb (hunk 2 above)
Comment (Jeko0) [reply week5_sub_strings/lib/sub_strings.rb]: removed long line

File: week5_sub_strings/spec/spec_helper.rb [hunk 3]
```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
```
Comment (alexshvedko) [inline week5_sub_strings/spec/spec_helper.rb]: i have written about commented lines in the channel

Comment (Jeko0) [reply week5_sub_strings/spec/spec_helper.rb]: removed commented lines

**Pull request merged**